libs/ledger/src/upow/synergetic_executor.cpp

```diff
@@ -51,7 +51,7 @@ void SynergeticExecutor::Verify(WorkQueue &solutions, ProblemData const &problem
       auto const &contract_address = solution->contract_digest();
 
       // create the contract
-      contract = factory_.Create(contract_address);
+      contract = factory_.Create(contract_address.ToHex());
```

Comment:
This looks a little confusing here because we have `contract_address` which isn't the new contract address it is an alias for `contract_digest`.

I also think that we should rework the interface of this factory so that it always expects a 32 byte digest (raw) and that internally it does the `.ToHex()`